### PR TITLE
handle 'Service unavailable' responses

### LINKED ISSUE #145: "Service Unavailable" response not raised properly
Hi there,

I found out a breach in the code / API. It looks like the Ads Insights API (and possibly other endpoints) can return a `Service Unavailable` in the response body with a `200` status code.

#### Symptoms
First of all, here is what happened in our code:
```python
# fb_campaign is properly initialized with an id, the api too
stats = fb_campaign.get_insights(fields=['actions'], params={
    'time_range': {
        'since': '2016-01-22', 
        'until': '2016-01-29'
    }
})
```
and the backtrace:
```
TypeError: string indices must be integers
  File "xxxxxx", line 147, in xxxxx
    stats = fb_campaign.get_insights(fields=['actions'], params=params)
  File "facebookads/objects.py", line 1283, in get_insights
    include_summary=False
  File "facebookads/objects.py", line 795, in iterate_edge_async
    include_summary=include_summary
  File "facebookads/objects.py", line 766, in iterate_edge
    iterator.load_next_page()
  File "facebookads/objects.py", line 163, in load_next_page
    self._queue = self.build_objects_from_response(response)
  File "facebookads/objects.py", line 181, in build_objects_from_response
    obj._set_data(data)
  File "facebookads/objects.py", line 444, in _set_data
    self._data[key] = data[key]
```
#### How to fix

[Here](https://github.com/facebook/facebook-python-ads-sdk/blob/2.5.1/facebookads/api.py#L96-L98) there should be a check on the response body, to verify that it's a string and doesn't contains "Service Unavailable". It's a bit dirty and I think you should report this issue internally on your API team. Also, we have this issue several times a week (not too often though), I think it could refer to one of your maintenances.

LMK if you want me to provide a PR to fix it quickly.

### PR DESCRIPTION
**Pull request by m-vdb**
fixes #145

Hi @rituparnamukherjee here is a small pull request to improve the error handling. Can you take a look? thanks :)

### CODE REVIEW

File: facebookads/test/unit.py [hunk 1]
```diff
@@ -442,5 +442,17 @@ def test_api_version_is_pulled(self):
...
+
+class FacebookResponseTestCase(unittest.TestCase):
+
+    def test_is_success_200(self):
+        resp = api.FacebookResponse(http_status=200)
+        self.assertTrue(resp.is_success())
+
+    def test_is_success_service_unavailable(self):
+        resp = api.FacebookResponse(body="Service Unavailable", http_status=200)
+        self.assertTrue(resp.is_success())
```
Comment (adongy) [inline facebookads/test/unit.py]: `self.assertFalse` instead of `True` ?

**Comment by m-vdb** [comment]
@rituparnamukherjee looks like the build is already failing on master for python3. Do you think you can merge this fix soon?

**Comment by rituparnamukherjee** [comment]
Yes, this will be fixed on master tomorrow.

**Comment by m-vdb** [comment]
@rituparnamukherjee please read the PR: this code was not made in order to fix the tests on master, this code was made to fix another issue... Can you reopen and merge when you deem it right?

**Comment by rituparnamukherjee** [comment]
Yep, I know the PR fixes another issue. I will be pushing some code changes tomorrow which will fix the build. Will take a look at the PR for merge post that.

**Comment by yoanthiebault** [comment]
@rituparnamukherjee, will this PR be merged or not ? I am facing this bug also and I rely on this PR to cleanly solve it :)

**Comment by m-vdb** [comment]
fixed by this [commit|https://github.com/facebook/facebook-python-ads-sdk/commit/04e0c683bf9119e3719d1a875d5d40ba8611776a]

**Pull request closed without merging**